plugins/lookup-table/src/test/java/pl/edu/uj/kimage/plugins/lookup/table/InvertFlowStepTest.java

```diff
@@ -0,0 +1,52 @@
+package pl.edu.uj.kimage.plugins.lookup.table;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import pl.edu.uj.kimage.api.Step;
+import pl.edu.uj.kimage.api.StepDependency;
+import pl.edu.uj.kimage.eventbus.EventBus;
+import pl.edu.uj.kimage.plugin.ImageCalculated;
+import pl.edu.uj.kimage.plugin.StepResultEvent;
+import pl.edu.uj.kimage.plugin.model.Color;
+import pl.edu.uj.kimage.plugin.model.Image;
+import pl.edu.uj.kimage.plugin.model.ImageBuilder;
+
+import java.util.ArrayList;
+
+import static org.mockito.Mockito.mock;
+
+public class InvertFlowStepTest {
+    private static final int ONE_PIXEL = 1;
+    private static final int STEP_NUMBER = 1;
+    private static final int Y_COORDINATE = 0;
+    private static final int X_COORDINATE = 0;
+    private EventBus eventBus;
+    private Image whiteImage;
+    private Image blackImage;
+
+    @Before
+    public void setUpd() {
+        blackImage = new ImageBuilder(ONE_PIXEL, ONE_PIXEL).withColor(Color.BLACK, X_COORDINATE, Y_COORDINATE).build();
+        whiteImage = new ImageBuilder(ONE_PIXEL, ONE_PIXEL).withColor(Color.WHITE, X_COORDINATE, Y_COORDINATE).build();
+        eventBus = mock(EventBus.class);
```

Comment:
Incorrect. When using JUnit you are not coupling yourself with internal implementation of used methods. However, you are doing so using mock the way you did. With JUnit you are just expecting that it will calculate certain things but you don't care how. If these methods won't exist in newer version you will easily know it. With mocks all these cases are not valid as someone, who is introducing modifications will learn it hard way (everything will compile correctly, maybe with some additional warnings for @Depricated, but behave differently). And instead of fixing one place, he/she will need to fix all the unit test for plugins. Therefore it is beneficial to provide test implementation which will follow some clear conventions and when changes are introduce you just adjust this single convention to the changes.

Since there is easy to avoid doing it, why wouldn't you do so?